zkevm-circuits/src/rlp_circuit_fsm.rs

```diff
@@ -2256,6 +2226,227 @@ impl<F: Field> RlpCircuitConfig<F> {
+                    is_decode_tag_start(meta),
+                    meta.query_advice(byte_value, Rotation::cur()) - 192.expr(),
+                    meta.query_advice(tag_value_acc, Rotation::cur()),
+                ),
+                0.expr(),
+            ];
+            let table_exprs = vec![
+                meta.query_advice(rlp_decoding_table.tx_id, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.format, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.byte_idx, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.depth, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.is_stack_push, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.value, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.value_prev, Rotation::cur()),
+            ];
+            input_exprs
+                .into_iter()
+                .zip(table_exprs)
+                .map(|(input, table)| (input * enable.expr(), table))
+                .collect()
+        });
+
+        meta.lookup_any("Decoding table stack op POP correspondence", |meta| {
+            let enable = select::expr(
+                sum::expr([
+                    meta.query_advice(rlp_decoding_table.is_stack_init, Rotation::next()),
+                    is_padding_in_dt.expr(Rotation::next())(meta),
+                ]),
+                0.expr(),
+                meta.query_advice(is_pop_op_lookup, Rotation::cur()),
+            );
+
+            let input_exprs = vec![
+                meta.query_advice(tx_id, Rotation::cur()),
+                meta.query_advice(format, Rotation::cur()),
+                meta.query_advice(byte_idx, Rotation::cur()),
+                meta.query_advice(depth, Rotation::cur()) - 1.expr(),
+                1.expr(),
+            ];
+            let table_exprs = vec![
+                meta.query_advice(rlp_decoding_table.tx_id, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.format, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.byte_idx, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.depth, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.is_stack_pop, Rotation::cur()),
+            ];
+            input_exprs
+                .into_iter()
+                .zip(table_exprs)
+                .map(|(input, table)| (input * enable.expr(), table))
+                .collect()
+        });
+
+        meta.lookup_any("Decoding table stack op UPDATE correspondence", |meta| {
+            let enable = meta.query_advice(is_update_op_lookup, Rotation::cur());
+
+            let input_exprs = vec![
+                meta.query_advice(tx_id, Rotation::cur()),
+                meta.query_advice(format, Rotation::cur()),
+                meta.query_advice(byte_idx, Rotation::cur()),
+                meta.query_advice(depth, Rotation::cur()),
```

Comment:
Should you add `al_idx` in the input exprs?

Reply:
`al_idx` is now part of input_exprs.